pkg/controller/endpointslice/reconciler.go

```diff
@@ -287,7 +289,10 @@ func (r *reconciler) reconcileByPortMapping(
 			}
 		}
 
-		// If an endpoint was updated or removed, mark for update or delete
+		// generate the slice labels and check if parent labels have changed
+		labels, svcLabelsUpdated := setEndpointSliceLabels(existingSlice, service)
```

Comment:
`svcLabelsUpdated` feels confusing here, like we changed the actual Service labels. Maybe `labelsChanged` could work instead?

Reply:
yep